graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -569,16 +569,15 @@ def _try_get_base_connections(self, class_name, class_name_to_definition,
             links[property_definition['name']].add(property_definition['linkedClass'])
 
         for end_direction, linked_classes in six.iteritems(links):
+            # The linked_classes set is the complete set of superclasses that a class must
+            # inherit from in order to be allowed in the edge end specified by end_direction.
+            # The base connection of an edge's end is the superclass of all classes allowed in the
+            # edge's end. Therefore, the base connection of the end specified by end_direction,
+            # if it exists, must be the class in linked_classes that is a subclass of all other
+            # classes in linked_classes.
             for linked_class in linked_classes:
-                subclass_set = self._subclass_sets[linked_class]
-                if len(linked_classes.intersection(subclass_set)) == 1:
-                    base_direction_connection = base_connections.get(end_direction, None)
-                    if base_direction_connection and base_direction_connection != linked_class:
-                        raise AssertionError(u'There already exists class "{}" in addition '
-                                             u'to class "{}" which is a subclass of all '
-                                             u'{} properties for class "{}".'
-                                             .format(base_direction_connection,
-                                                     linked_class, end_direction, class_name))
+                inheritance_set = self._subclass_sets[linked_class]
```

Comment:
why change the name to something more ambiguous?

Reply:
This seems to be an uncaught bug. Let me add a test for it.